cmd/build.go

```diff
@@ -77,151 +78,145 @@ var buildLinuxCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
-		assertHoverInitialized()
-		packaging.WindowsMsiPackagingTask.AssertInitialized()
-		packaging.AssertDockerInstalled()
-
-		buildNormal("windows", nil)
-		packaging.WindowsMsiPackagingTask.Pack(buildVersion)
+		subcommandBuild("windows", packaging.WindowsMsiTask)
 	},
 }
 
-// checkForMainDesktop checks and adds the lib/main_desktop.dart dart entry
-// point if needed
-func checkForMainDesktop() {
-	if buildTarget != "lib/main_desktop.dart" {
-		return
+// TODO: replace targetOS with a same Task type for build (build.Task) ?
+func subcommandBuild(targetOS string, packagingTask packaging.Task) {
+	assertHoverInitialized()
+	packagingTask.AssertInitialized()
+
+	if !buildSkipFlutterBuildBundle {
+		cleanBuildOutputsDir(targetOS)
+		buildFlutterBundle(targetOS)
 	}
-	_, err := os.Stat("lib/main_desktop.dart")
+	if buildDocker {
+		var buildFlags []string
+		buildFlags = append(buildFlags, commonFlags()...)
+		buildFlags = append(buildFlags, "--skip-flutter-build-bundle")
+		buildFlags = append(buildFlags, "--skip-engine-download")
+		if buildVersionNumber != "" {
+			buildFlags = append(buildFlags, "--version-number", buildVersionNumber)
+		}
+		if buildDebug {
+			buildFlags = append(buildFlags, "--debug")
+		}
+		dockerHoverBuild(targetOS, packagingTask, buildFlags, nil)
+	} else {
+		buildGoBinary(targetOS, nil)
+		packagingTask.Pack(buildVersionNumber)
+	}
+}
+
+func commonFlags() []string {
+	f := []string{}
+	if buildTarget != config.BuildTargetDefault {
+		f = append(f, "--target", buildTarget)
+	}
+	if buildGoFlutterBranch != config.BuildBranchDefault {
+		f = append(f, "--branch", buildGoFlutterBranch)
+	}
+	if buildOpenGlVersion != config.BuildOpenGlVersionDefault {
+		f = append(f, "--opengl", buildOpenGlVersion)
+	}
+	return f
+}
+
+// assertTargetFileExists checks and adds the lib/main_desktop.dart dart entry
+// point if needed
+func assertTargetFileExists(targetFilename string) {
+	_, err := os.Stat(targetFilename)
 	if os.IsNotExist(err) {
-		log.Warnf("Target file \"lib/main_desktop.dart\" not found.")
-		log.Warnf("Let hover add the \"lib/main_desktop.dart\" file? ")
-		if askForConfirmation() {
-			fileutils.CopyAsset("app/main_desktop.dart", filepath.Join("lib", "main_desktop.dart"), fileutils.AssetsBox)
-			log.Infof("Target file \"lib/main_desktop.dart\" has been created.")
-			log.Infof("       Depending on your project, you might want to tweak it.")
-			return
+		log.Warnf("Target file \"%s\" not found.", targetFilename)
+		if targetFilename == config.BuildTargetDefault {
+			log.Warnf("Let hover add the \"lib/main_desktop.dart\" file? ")
+			if askForConfirmation() {
+				fileutils.CopyAsset("app/main_desktop.dart", filepath.Join("lib", "main_desktop.dart"), fileutils.AssetsBox)
+				log.Infof("Target file \"lib/main_desktop.dart\" has been created.")
+				log.Infof("       Depending on your project, you might want to tweak it.")
+				return
+			}
 		}
```

Comment:
If the target file isn't found, why can't we create it from the template with the name from the target file? Currently it only works if the tartget file name is the default name, but I see no reason why it shouldn't work for any target file name.

Reply:
Doing this (and having user interaction from a build command) is already a bit odd. In general I'd like to keep commands clean. If build cannot continue, it should report an error and maybe suggest a way to fix it.

Now at the same time I'd like to make using hover easy. So this is a great convenience feature for new users. But if a user wants to build with a file other than `lib/main_desktop.dart`, they should know what you're doing and be able to fix it manually.